### ISSUE TITLE
Acting Pay More than a Year

### ISSUE THREAD

**Issue by pducharme** [andrewnordlund/backpayCalc#70]
Hi! Nice calculator BTW :) 

IT02-Step 8:   Since When I joined in 2018 (was in a different CA before since 2003)
IT02-Step 8 :  On Dec. 22, 2020 --> Got the pay upgrade for that year in the CA.
IT02-Step 8 :  Was already at Top Level, so no step increase in 2021 until first Acting.
IT03-Step 4 :  (Acting) from 2021-03-01 to 2022-03-01 (1Y Acting, was in fact 1Y and 16 days)
IT03-Step 5 :  (Acting) from 2022-03-01 to 2022-03-17 (16 days - the end of the 1Y Acting I had this step Increase)
IT02-Step 8 :  Back to IT-02 on 2022-03-18.
IT03-Step 4 :  (Acting) from 2022-05-30 to 2023-05-30 (1Y) -- Yes, lost the Step 5 because of a break in the acting.
IT03-Step 5 : Same day of my 1Y acting finishing, I got Promoted to IT-03 Substantive (at Step 5).

Calculator wasn't understanding it :)

**Comment by andrewnordlund** [comment]
What do you mean "Calculator wasn't understanding it"?

What happened?  Did it just do nothing when you clicked "Calculate"?  Did you get an understandable, but knowingly false result?   When you joined in 2018, what step did you start with?

When filing a bug report like this, it's helpful to provide the steps you took to produce the error, then a description of Expected Result and Actual result.  (See Issue #66 for example.)

**Comment by pducharme** [comment]
Hi! Sorry I wasn't clear.  When I joined in 2018, i was alreday Top Level (Step 8) IT-02.  I came from CS-02 Top Level at CRA prior to join SSC.  I started in 2003 at CRA.

If you click that link you'll see in the table : https://nordburg.ca/calculators/backpayCalc/2021/backpayCalc.html?lvl=2&stp=7&strtdt=2010-02-01&enddt=2024-03-17&pdate0=2023-05-30&plvl0=3&afrom0=2021-12-22&ato0=2022-03-17&alvl0=3&afrom1=2022-05-30&ato1=2023-05-29&alvl1=3

The dates are mixed up and it see Step 6 IT-03 but that never happened, and he also never put IT-03 Step 5 after the full Year at Step 4 in the acting.

**Comment by pducharme** [comment]
Hi!  In case you haven't seen yet, the Breakdown by year is now known : 

2021: 3
2022: 4,75
2023 : 3,5
2024 : 2,25

source : [PIPSC Website](https://pipsc.ca/groups/cs/it-tentative-agreement-update?fbclid=IwAR2M88zFsVrKnhQ3mG9NUfHwD5p3spl9BH10yULoCUygNzArPoTEB6yuDqE)

**Comment by andrewnordlund** [comment]
In summary:

The calculator does not give an anniversary raises for Actings that last more than 1 year.

Also, ensure that Actings for more than 1 year which start before the beginning of the TA/CA are accounted for.

**Comment by andrewnordlund** [comment]
I think I have a fix. 

It took a bit of work to account for Actings that start before the start of the TA that last for more than a year.

As for promotions that happen the day after an acting end; just set the day of the promotion to the first day of that acting.

Example:
You acted IT-03 from 2022-05-30 to 2023-05-30, but were promoted on 2023-05-30 to substantive IT-03.

Therefore, as I understand it, you will move from IT-03 Step 5 to Step 6 on 2024-05-30.

If I am correct, then for the purposes of _retro-pay_, that's the same thing as simply being promoted on 2022-05-30.

Take a look at Dev, and let me know if it's correct:
https://dev.nordburg.ca/calculators/backpayCalc/2021/backpayCalc.html?lvl=2&stp=7&strtdt=2010-02-01&enddt=2024-03-17&pdate0=2022-05-30&plvl0=3&afrom0=2021-03-01&ato0=2022-03-17&alvl0=3

**Comment by pducharme** [comment]
Hi! it certainly look better.  It seems to work if I don't put my 2nd Acting and use the date of the start of the 2nd acting as my promotion date.  It's still different results from some spreadsheets, but it's within 130$, so I can say it gives a good idea of the Retropay.

**Comment by andrewnordlund** [comment]
If it's the same spreadsheet that I've tried there are two reasons for this that I can see:
1. The spreadsheet calculates days with the `NETWORKDAYS()` function which calculates working days, ie: total days - weekends - holidays.  But you get paid for holidays.  (I also suspect it uses US holidays.)  This calculator, however, just goes from day to day to day, checking if it's a weekend or not.  If not, it adds one day's pay to the running total.
2. This calculator starts with the base pay from the last CA; then calculates the next raise based on the percentage increase.  It rounds it to the nearest cent.  Then for the next increase it does the same.  I think the spreadsheet creates the tables for each pay period and rounds them to the nearest dollar and uses those.    I seem to remember reading somewhere how TBS actually does the calculation.  It rounds the annual salaries up, down, or to the nearest.  If I can find that, I'll update this calculator.

Anyway, I've pushed this to prod, so I'll close this Issue.
